clients/admin-ui/src/features/system/system-form-declaration-tab/PrivacyDeclarationDisplayGroup.tsx

```diff
@@ -27,37 +29,59 @@ const PrivacyDeclarationRow = ({
   title?: string;
   handleDelete?: (dec: PrivacyDeclarationResponse) => void;
   handleEdit: (dec: PrivacyDeclarationResponse) => void;
-}) => (
-  <>
-    <Box px={6} py={4}>
-      <HStack>
-        <LinkBox
-          onClick={() => handleEdit(declaration)}
-          w="100%"
-          h="100%"
-          cursor="pointer"
-        >
-          <LinkOverlay>
-            <Text>{title || declaration.data_use}</Text>
-          </LinkOverlay>
-        </LinkBox>
-        <Spacer />
-        {handleDelete ? (
-          <IconButton
-            aria-label="delete-declaration"
-            variant="outline"
-            zIndex={2}
-            size="sm"
-            onClick={() => handleDelete(declaration)}
+}) => {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  return (
+    <>
+      <Box px={6} py={4} data-testid={`row-${declaration.data_use}`}>
+        <HStack>
+          <LinkBox
+            onClick={() => handleEdit(declaration)}
+            w="100%"
+            h="100%"
+            cursor="pointer"
           >
-            <DeleteIcon />
-          </IconButton>
-        ) : null}
-      </HStack>
-    </Box>
-    <Divider />
-  </>
-);
+            <LinkOverlay>
+              <Text>{title || declaration.data_use}</Text>
+            </LinkOverlay>
+          </LinkBox>
+          <Spacer />
+          {handleDelete ? (
+            <>
+              <IconButton
+                aria-label="delete-declaration"
+                variant="outline"
+                zIndex={2}
+                size="sm"
+                onClick={onOpen}
+                data-testid="delete-btn"
+              >
+                <DeleteIcon />
+              </IconButton>
+              <ConfirmationModal
+                isOpen={isOpen}
+                onClose={onClose}
+                onConfirm={() => handleDelete(declaration)}
+                title="Delete data use declaration"
+                message={
+                  <Text>
+                    You are about to delete the data use declaration{" "}
+                    <Text color="complimentary.500" as="span" fontWeight="bold">
+                      {title || declaration.data_use}
+                    </Text>
+                    , including all its cookies. Are you sure you want to
+                    continue?
+                  </Text>
```

Comment:
I didn't see a design in the original ticket, but our modals have been skewing "centered" on the page with the `isCentered` prop